### PR TITLE
Implements access token

### PR DESCRIPTION
**Pull request by BaltaJmn**

### CODE REVIEW

File: core/auth/build.gradle.kts [hunk 1]
```diff
@@ -8,6 +8,7 @@ android {
     defaultConfig {
         buildConfigField("String", "CLIENT_ID", project.property("CLIENT_ID").toString())
         buildConfigField("String", "CAMPAIGN_ID", project.property("CAMPAIGN_ID").toString())
+        buildConfigField("String", "CLIENT_BASE_64", project.property("CLIENT_BASE_64").toString())
```
Comment (Naxomiun) [inline core/auth/build.gradle.kts:11]: Is this necessary? can't we transform it to base64 at runtime?

File: core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt [hunk 2]
```diff
@@ -2,28 +2,50 @@ package com.wachon.spotiwrap.core.network.di
...
 private const val BASE_URL: String = "https://api.spotify.com"
 
 val NetworkModule: Module
     get() = module {
         factory { ThreadInterceptor() }
+        factory { NetworkInterceptor(get()) }
         factory { TokenInterceptor(get()) }
-        factory {
+
+        single(named("NetworkModule")) {
```
Comment (Naxomiun) [inline core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt]: I can think of some more understandable names:
- BasicAuthClient
- AuthClient 
...etc

File: core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt [hunk 3]
```diff
@@ -2,28 +2,50 @@ package com.wachon.spotiwrap.core.network.di
...
             HttpClient.getClient(
                 baseUrl = BASE_URL,
+                interceptors = listOf(
+                    get<ThreadInterceptor>(),
+                    get<NetworkInterceptor>()
+                )
+            )
+        }
+
+        single(named("TokenModule")) {
```
Comment (Naxomiun) [inline core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt]: Same than above. With that naming I wouldn't know which client to use

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -5,7 +5,7 @@ compileSdk = "33"
 targetSdk = "33"
 minSdk = "24"
 kotlin = "1.8.10"
-gradle = "8.0.0"
+gradle = "7.4.2"
```
Comment (Naxomiun) [inline gradle/libs.versions.toml]: 😢

File: features/login/src/main/kotlin/com/wachon/spotiwrap/features/login/domain/GetAccessToken.kt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.wachon.spotiwrap.features.login.domain
+
+import com.wachon.spotiwrap.core.auth.scopes.GetAuthConfigUseCase
+import com.wachon.spotiwrap.core.common.model.TokenResponseModel
+import com.wachon.spotiwrap.features.login.data.TokenRepository
+
+interface GetAccessTokenUseCase {
```
Comment (Naxomiun) [inline features/login/src/main/kotlin/com/wachon/spotiwrap/features/login/domain/GetAccessToken.kt:7]: This is not more related to Auth module than login one?

File: features/login/src/main/kotlin/com/wachon/spotiwrap/features/login/domain/RefreshToken.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.wachon.spotiwrap.features.login.domain
+
+import com.wachon.spotiwrap.core.auth.token.GetRefreshTokenUseCase
+import com.wachon.spotiwrap.core.common.model.TokenResponseModel
+import com.wachon.spotiwrap.features.login.data.TokenRepository
+
+interface RefreshTokenUseCase {
```
Comment (Naxomiun) [inline features/login/src/main/kotlin/com/wachon/spotiwrap/features/login/domain/RefreshToken.kt:7]: Same than above

File: core/auth/build.gradle.kts (hunk 1 above)
Comment (BaltaJmn) [reply core/auth/build.gradle.kts:11]: If we do it at runtime, we have to find a way to do it on android O version. Another thing is that in the Spotify Dashboard, you can refresh the campaign id, so we should change it here as well, so I don't think there's much of a problem in having it hardcoded.

File: core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt (hunk 2 above)
Comment (BaltaJmn) [reply core/network/src/main/kotlin/com/wachon/spotiwrap/core/network/di/NetworkModule.kt]: I will change the names to AuthClient and UserClient

File: gradle/libs.versions.toml (hunk 4 above)
Comment (BaltaJmn) [reply gradle/libs.versions.toml]: Solved, I upgraded my Android Studio
6f7

**Pull request merged**